### PR TITLE
Small fixes to get existing LDAP provider module working 

### PR DESCRIPTION
**Pull request by lullis**
Changes necessary to get #1181 working with the current LDAP Provider module.

### CODE REVIEW

File: synapse/util/ldap_auth_provider.py [hunk 1]
```diff
@@ -237,13 +237,14 @@ def _ldap_simple_bind(self, server, localpart, password):
...
             conn = ldap3.Connection(server, bind_dn, password,
-                                    authentication=ldap3.AUTH_SIMPLE)
+                                    authentication=ldap3.SIMPLE)
             logger.debug(
                 "Established LDAP connection in simple bind mode: %s",
                 conn
             )
 
             if self.ldap_start_tls:
+                conn.bind()
```
Comment (mweinelt) [inline synapse/util/ldap_auth_provider.py]: As I understand it that would mean to authenticate against the ldapd in plaintext.

File: synapse/util/ldap_auth_provider.py [hunk 2]
```diff
@@ -237,13 +237,14 @@ def _ldap_simple_bind(self, server, localpart, password):
                 base=self.ldap_base
             )
             conn = ldap3.Connection(server, bind_dn, password,
-                                    authentication=ldap3.AUTH_SIMPLE)
+                                    authentication=ldap3.SIMPLE)
```
Comment (mweinelt) [inline synapse/util/ldap_auth_provider.py]: This was addressed in the external module. Whether `SIMPLE` or `AUTH_SIMPLE` works for you is dependant on your ldap3 version.

**Comment by mweinelt** [comment]
For what it's worth, ldap changes should be requested against https://github.com/matrix-org/matrix-synapse-ldap3.

I don't think your pull request to make StartTLS work for you is the correct way, because you would send your credentials in plain, before switching over to TLS.

**Comment by ara4n** [comment]
it sounds like @mweinelt addressed these issues in matrix-org/matrix-synapse-ldap3.

**Pull request closed without merging**